### ISSUE TITLE
key.bin creation in mp4-dash.py?

### ISSUE THREAD

**Issue by SimoneCarnio** [axiomatic-systems/Bento4#710]
Hello,
there is a specific reason for not adding the --output-encryption-key option also to mp4-dash.py as in mp4-hls.py ?

I'm referring to these lines:

https://github.com/axiomatic-systems/Bento4/blob/d02ef8230a8fee4904a750eb912521c5f1c74e0b/Source/Python/utils/mp4-hls.py#L254-L256

I trying to use mp4-dash.py instead of mp4-hls.py to integrate subtitles management and this option would make the scripts migration easy. 

For testing purpose I'm using the  --hls-key-url  to point to a url with a test key and all works correct, just wondering if I'm missing the reason for not having the same behavior

Thanks

**Comment by lfaureyt** [comment]
The HLS specification states that the encryption key should be available through a URI and defines the corresponding syntax (`URI` attribute of the `#EXT-X-KEY` tag), which allows simple and straightforward applications like dropping the key into a binary file and making it available through a `file://key.bin` URL, hence this feature of `mp4-hls.py` , I guess.

AFAIK, the DASH standard does not specify anything equivalent. It specifies encryption key material identification through 16-byte keyIds and signalization (through PSSH descriptors) but leaves it to proprietary key systems to determine how to retrieve key values from keyIds.

**Comment by SimoneCarnio** [comment]
Yes but when you use the --hls option the script also create the hls playlist, so I think it should create also the file as mp4-hls.py do

**Comment by lfaureyt** [comment]
OK, since `mp4-dash.py` can actually produce HLS output by activating the `--hls` option, this actually makes sense.

**Comment by MarianoFacundoArch** [comment]
Upvote for this... It should really be added, couldn't find a way of doing so

**Comment by SimoneCarnio** [comment]
@MarianoFacundoArch  if you want to try, this is a modified version of the script that I was using for my use case, maybe it's ok also for you. 
Please note that I've ended up to removed the hls encryption for the moment form my workflow because of some incompatibilities with some players with the fMP4 encapsulation.
